### PR TITLE
Remove lint rule to prevent global Promise

### PR DESCRIPTION
**Pull request by snewcomer**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ module.exports = {
     'no-console': 'error', // no longer recommended in eslint v6, this restores it
     'no-eq-null': 'error',
     'no-eval': 'error',
-    'no-restricted-globals': ['error', { name: 'Promise', message: 'Global Promise does not work in IE11' }],
+    'no-restricted-globals': ['error', { name: 'Promise', message: 'Must use RSVP.Promise' }],
```
Comment (snewcomer) [inline .eslintrc.js:55]: We will want to maintain this correct?  Or with IE11 support dropped, can we remove this lint rule @runspired

Comment (runspired) [reply .eslintrc.js:55]: with IE11 support dropped we can drop this

Comment (snewcomer) [reply .eslintrc.js]: Any reason we can't look to drop rsvp?  Need an RFC?

Comment (runspired) [reply .eslintrc.js]: We may still rely on hooking into it to allow promise cancellation in one situation.

**Pull request merged**